services/Donation.js

```diff
@@ -60,6 +60,87 @@ class DonationService {
       return false;
     }
   }
+
+  /**
+   * @description get total donations
+   */
+  static async getTotal(startDate, endDate) {
+    try {
+      const matchQuery = {};
+      if (startDate & endDate) {
+        matchQuery.createdAt = { $gte: startDate, $lte: endDate };
+      }
+
+      const totalDonations = await Donation.aggregate([
+        { $match: matchQuery },
+        { $group: { _id: null, totalAmount: { $sum: "$transaction.amount" } } }
+      ]);
+
+      if (totalDonations.length === 0) {
+        return { totalAmount: 0 };
+      }
+
+      return { totalAmount: totalDonations[0].totalAmount };
+    } catch (e) {
+      throw new Error("Internal Server Error");
+    }
+  }
+
+  /**
+   * @description get total donations and donors
+   */
+  static async getTotalDonorsAndAmount() {
+    try {
+      const donationStats = await Donation.aggregate([
+        {
+          $group: {
+            _id: null,
+            totalAmount: { $sum: "$transaction.amount" },
+            uniqueDonorsCount: { $addToSet: "$donor.email" }
+          }
+        }
+      ]);
+
+      const result = {
+        totalAmount:
+          donationStats.length > 0 ? donationStats[0].totalAmount : 0,
+        uniqueDonorsCount:
+          donationStats.length > 0
+            ? donationStats[0].uniqueDonorsCount.length
+            : 0
+      };
+
+      return result;
+    } catch (e) {
+      throw new Error("Internal Server Error");
```

Comment:
See my comment above about how we should log the error and rethrow it. 
Please make the change here as well.